runtime-sdk/src/modules/rofl/app/registration.rs

```diff
@@ -191,4 +198,49 @@ where
 
         Ok(())
     }
+
+    async fn collect_provider_metadata(
+        &self,
+        metadata: &mut BTreeMap<String, String>,
+    ) -> Result<()> {
+        let rsp = self
+            .env
+            .host()
+            .attestation()
+            .attest_labels(AttestLabelsRequest {
+                labels: vec![LABEL_PROVIDER.to_string()],
+            })
+            .await?;
+
+        // Decode the attestation to check if the provider label is set and skip setting
+        // metadata in case it is not.
+        let la: LabelAttestation = cbor::from_slice(&rsp.attestation)?;
+        if la.labels.get(LABEL_PROVIDER) == Some(&String::new()) {
```

Comment:
Why don't we use `contains_key`? Are we sure that the label is always present and cannot be `None`?

Reply:
If the node is corrupted, it could not attest an empty label and in that case this could be `None`, right?

Reply:
You are right, but I would still check, so that it is in line with the comment `// Decode the attestation to check if the provider label is set and skip setting metadata in case it is not.`.

Reply:
Fixed.